client/extra.go

```diff
@@ -6,17 +6,22 @@ import (
 	"github.com/ovrclk/akash/x/market/types"
 )
 
-// TODO: implement with search parameters
-
-func (c *qclient) ActiveLeasesForProvider(id sdk.AccAddress) (mquery.Leases, error) {
-	leases, err := c.Leases(mquery.LeaseFilters{})
+// ActiveLeasesForProvider filter the Lease search to active state and then
+// provider's ID AccAddress.
+func (c *qclient) ActiveLeasesForProvider(provider sdk.AccAddress) (mquery.Leases, error) {
+	// NOTE: Currently unable to filter by Provider in tendermint scan
+	lf := mquery.LeaseFilters{
+		// TODO: Add filters when tests prove functionality
+	}
+	leases, err := c.Leases(lf)
 	if err != nil {
 		return nil, err
 	}
 
+	// TODO: Secondary filtering *should* not be necessary
 	var filtered mquery.Leases
 	for _, lease := range leases {
-		if lease.Provider.Equals(id) && lease.State == types.LeaseActive {
+		if lease.Provider.Equals(provider) && lease.State == types.LeaseActive {
```

Comment:
just leave this stuff as it was - these changes don't really add anything.

/bump #725 #604